amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/model/rma/type/RecordCategoryType.java

```diff
@@ -204,6 +218,50 @@ public Void doWork()
 
     }
 
+    /**
+     * Record Category move behaviour
+     *
+     * @see org.alfresco.repo.node.NodeServicePolicies.OnMoveNodePolicy#onMoveNode(org.alfresco.service.cmr.repository.ChildAssociationRef, org.alfresco.service.cmr.repository.ChildAssociationRef)
+     */
+    @Override
+    @Behaviour
+            (
+                    kind = BehaviourKind.CLASS,
+                    notificationFrequency = NotificationFrequency.FIRST_EVENT
+            )
+    public void onMoveNode(ChildAssociationRef oldChildAssocRef, ChildAssociationRef newChildAssocRef)
+    {
+        if (nodeService.getType(newChildAssocRef.getChildRef()).equals(TYPE_RECORD_CATEGORY) && dispositionService.getDispositionSchedule(oldChildAssocRef.getParentRef()) != null)
+        {
+            for (ChildAssociationRef newChildRef : nodeService.getChildAssocs(newChildAssocRef.getChildRef(),
+                    ContentModel.ASSOC_CONTAINS,
+                    RegexQNamePattern.MATCH_ALL))
+            {
+                final NodeRef newNodeRef = newChildRef.getChildRef();
+
+                AuthenticationUtil.runAs(() -> {
+                    // clean record folder
+                    cleanDisposableItem(nodeService, newNodeRef);
+
+                    // re-initialise the record folder
+                    recordFolderService.setupRecordFolder(newNodeRef);
+
+                    // sort out the child records
+                    for (NodeRef record : recordService.getRecords(newNodeRef))
+                    {
+                        // clean record
+                        cleanDisposableItem(nodeService, record);
+
+                        // Re-initiate the records in the new folder.
+                        recordService.file(record);
+                    }
+
+                    return null;
+                }, AuthenticationUtil.getSystemUserName());
+            }
+        }
+    }
+
```

Comment:
Will move some common code to AbstractDisposableItem, will have to do one more commit.